src/main/java/seedu/address/model/util/Date.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.model.util;
+
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a Date in Sharkie.
+ * Guarantees: immutable; date is valid as declared in {@link #isValidDate(String)}
+ */
+
+public class Date {
+
+    public static final String DATE_FORMAT = "dd/MM/yyyy";
+    public static final String MESSAGE_CONSTRAINTS = "Dates should be in form: " + DATE_FORMAT;
+
+    private LocalDate date;
+
+    /**
+     * Constructs a valid Date object.
+     * @param date A valid date.
+     */
+    public Date(LocalDate date) {
+        this.date = date;
+    }
+
+    /**
+     * Returns true if a given String is a valid date format.
+     */
+    public static boolean isValidDate(String test) {
+        try {
+            SimpleDateFormat format = new SimpleDateFormat(DATE_FORMAT);
```

Comment:
Is there a reason to use SimpleDateFormat here and DateTimeFormatter below? From what I google (not v sure) it seems SimpleDateFormat is somewhat legacy and DateTimeFormatter is favoured nowadays. Also they may not have identical patterns

https://stackoverflow.com/questions/36722873/datetimeformatter-not-compatible-with-simpledateformat